src/cdk/testing/protractor/protractor-element.ts

```diff
@@ -6,7 +6,7 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {ModifierKeys} from '@angular/cdk/testing';
+import {ModifierKeys} from '@angular/cdk/private/testing';
```

Comment:
Is it ok to have files in our published packages refer to the unpublished `@angular/cdk/private/testing` package? It feels like it might result in a broken release

Reply:
Yeah, I was going to comment on that as well. This won't work since we release CDK testing but not the private stuff.

Not surr what the ideal solution would be. I think we decided to keep them in cdk testing, but not export them in the public API. The private/testing entrypoint then just re-exports the utilities from cdk/testing/internal.ts (as an example)

Reply:
Why not release the `private` package then? We can still build and release it, just not mention it in the docs and then if people stumble upon it, it has `private` in its name so they'd be using it at their own risk. One way or another we have to export these symbols through _somewhere_ so that the existing tests can use them. 

Exposing them through something like `cdk/testing/internal` seems like it would achieve the same, except that now we'll have two "private sounding" packages (`testing/internal` and `private/testing`) which can be confusing and it can still show up when people are auto-importing. 

Another approach can be to prefix the names with `ɵ` like in core, but it'll be more annoying for us to use.

Reply:
I think what @devversion mean by `internal` is like an equivalent of `public-api.ts` but for internal utilities. Any internal utilities would be exported through `internal.ts` and not through `public-api.ts`. `internal.ts` could then have a separate build rule for `@angular/cdk/private/testing` to import from and re-export the internal symbols

Reply:
Yeah, that's not really my idea, but I think that was the one of the proposals in the slack channel. My initial feeling always was that we can just export these utils as part of the public entry-point and either prefix them or just expose them.

Since we rely on these utils from within the harnesses, they should be reliable enough to be exposed in the `cdk/testing` entry-point too?

Reply:
I think we didn't want to make `@angular/cdk/testing` a dumping ground for stuff that just happened to wind up as part of the public API because we needed it in our tests.

cc @jelbourn

Reply:
I agree, but apparently we don't need it just for tests anymore. i.e. we use it in the harnesses and assume that it is reliable enough to be shipped.

Reply:
Yeah, I definitely don't want to publish any sort of `private` or `internal` package. People will ignore the name and use it anyway. My proposal was for the functions to live in `cdk/testing`, not export them through `public-api`, and then re-export them through cdk/private which we don't publish

Reply:
So to be clear, we want to have `cdk/testing/private` for these? What about the handful of APIs that are in `cdk/private/testing`? That feels like it could get confusing.

Reply:
I don't care where they live in `cdk/testing`, so long as they're not part of `public-api.ts`

Reply:
My question is whether we should get rid of `cdk/private/testing` since we're moving the fake events into `cdk/testing/private` (note the subtle difference).

Reply:
I think the idea was to _only_ move the event utilities into `cdk/testing/private.ts` as a "somewhat" workaround to get them shipped privately while still having them in `cdk/private/testing`.

Moving everything back into `cdk/testing` for consistency will mean that we ship unnecessary stuff into the `cdk/testing` bundles, but in general I agree that it is more clean if we just have something like `cdk/testing` (public) and `cdk/testing/private` (this time a folder).

Reply:
Within our own code we have have e.g. `cdk/testing/fake-events` so long as it's export publicly exported.